Libraries/Dijkstra/Dijkstra.h

```diff
@@ -0,0 +1,262 @@
+      if (tiles_map.getTile(current_coord->x + 1, current_coord->y).getVisited() == true && matrix[current_coord->x + 1][current_coord->y].checked == false) {
+        open_vector.pushTop(&matrix[current_coord->x + 1][current_coord->y].current);
+        matrix[current_coord->x + 1][current_coord->y].checked = true;
+      }
+      if (matrix[current_coord->x + 1][current_coord->y].weight > matrix[current_coord->x][current_coord->y].weight + tiles_map.getTile(current_coord->x + 1, current_coord->y).getWeight()) {
+        matrix[current_coord->x + 1][current_coord->y].weight = matrix[current_coord->x][current_coord->y].weight + tiles_map.getTile(current_coord->x + 1, current_coord->y).getWeight();
+        matrix[current_coord->x + 1][current_coord->y].prev = matrix[current_coord->x][current_coord->y].current;
+      }
+    }
+    if (tiles_map.getTile(current_coord->x, current_coord->y).getWest() == true && tiles_map.getTile(current_coord->x, current_coord->y - 1).getBlack() == false) {
+      if (tiles_map.getTile(current_coord->x, current_coord->y - 1).getVisited() == true && matrix[current_coord->x][current_coord->y - 1].checked == false) {
+        open_vector.pushTop(&matrix[current_coord->x][current_coord->y - 1].current);
+        matrix[current_coord->x][current_coord->y - 1].checked = true;
+      }
+      if (matrix[current_coord->x][current_coord->y - 1].weight > matrix[current_coord->x][current_coord->y].weight + tiles_map.getTile(current_coord->x, current_coord->y - 1).getWeight()) {
+        matrix[current_coord->x][current_coord->y - 1].weight = matrix[current_coord->x][current_coord->y].weight + tiles_map.getTile(current_coord->x, current_coord->y - 1).getWeight();
+        matrix[current_coord->x][current_coord->y - 1].prev = matrix[current_coord->x][current_coord->y].current;
+      }
+    }
+    open_vector.popFront();
+    
+  }
+
+  if (non_visited_vector.getSize() > 0) current_coord = non_visited_vector[0];
+  else Serial.print("No hay tiles por recorrer");
+
+  //Comparing all the non Visited Tiles
+  for (int i = 1 ; i < non_visited_vector.getSize() ; ++i) {
+    if (matrix[current_coord->x][current_coord->y].weight > matrix[non_visited_vector[i]->x][non_visited_vector[i]->y].weight) {
+      current_coord = non_visited_vector[i];
+    }
+  }
+
+  ///////////////////////////////////////////////////////////////////////////////
+  for (int i = 0 ; i < tiles_map.getRows() ; ++i) {
+    for (int j = 0 ; j < tiles_map.getColumns() ; ++j) {
+      Serial.print("(");
+      Serial.print(matrix[i][j].current.x);
+      Serial.print(",");
+      Serial.print(matrix[i][j].current.y);
+      Serial.print(")");
+      Serial.print("[");
+      Serial.print(matrix[i][j].prev.x);
+      Serial.print(",");
+      Serial.print(matrix[i][j].prev.y);
+      Serial.print("]");
+      Serial.print(matrix[i][j].weight);
+      Serial.print("  ");
+    }
+    Serial.println("");
+  }
+  Serial.println("");
```

Comment:
Make it a function

Reply:
Rule of thumb, 100 short and modular functions of 20 lines each is better than 20 functions with 100 lines each.